oxidation/libparsec/crates/types/src/regex.rs

```diff
@@ -26,40 +26,27 @@ impl PartialEq for Regex {
     }
 }
 
+#[cfg(test)]
 mod tests {
-    #[test]
-    fn test_one_wildcard_regex() {
-        let r = crate::regex::Regex::from_pattern("*.rs").expect("Should be valid");
-
-        assert!(r.0.is_match("foo.rs"));
-        assert!(!r.0.is_match("bar.py"));
-    }
-
-    #[test]
-    fn test_two_wildcard_regex() {
-        let r = crate::regex::Regex::from_pattern("*bar*").expect("Should be valid");
-
-        assert!(r.0.is_match("foobar.rs"));
-        assert!(r.0.is_match("bar"));
-        assert!(!r.0.is_match("foo.py"));
-    }
-
-    #[test]
-    fn test_escape_dollar_in_pattern() {
-        let r = crate::regex::Regex::from_pattern("$myf$le.*").expect("Should be valid");
-        assert!(r.0.is_match("$myf$le.txt"));
-        assert!(!r.0.is_match("bar"));
-    }
-
-    #[test]
-    fn test_regex_str_creation() {
-        let r = crate::regex::Regex::from_regex_str(r"fooo?$").expect("Should be valid");
-        assert!(r.0.is_match("foo"));
-        assert!(r.0.is_match("fooo"));
-        assert!(!r.0.is_match("fooooooo"));
-    }
-
-    #[test]
+    use rstest::rstest;
+
+    #[rstest]
+    #[case(crate::regex::Regex::from_pattern("*.rs").expect("Should be valid"), "foo.rs", "bar.py", "bar.rs")]
+    #[case(crate::regex::Regex::from_pattern("*bar*").expect("Should be valid"), "foobar.rs", "dummy", "foobarrrrr.py")]
+    #[case(crate::regex::Regex::from_pattern("$myf$le.*").expect("Should be valid"), "$myf$le.txt", "bar", "$myf$le.rs")]
+    #[case(crate::regex::Regex::from_regex_str(r"fooo?$").expect("Should be valid"), "foo", "fo", "fooo")]
```

Comment:
To simplify reading those case, move the compilation of the regex expr in the body of the test.